node/rest/src/routes.rs

```diff
@@ -334,6 +335,13 @@ impl<N: Network, C: ConsensusStorage<N>, R: Routing<N>> Rest<N, C, R> {
             return Err(RestError(format!("Unable to broadcast transaction '{}' (node is syncing)", fmt_id(tx.id()))));
         }
 
+        // If the transaction exceeds the transaction size limit, return an error.
```

Comment:
Technically, this function's signature takes in: `Json(tx): Json<Transaction<N>>` as argument, which can also bypass the size check there. Do we have any fundamental risk of an OOM from that attack vector?

Reply:
We limit request sizes here so we should be ok: https://github.com/AleoHQ/snarkOS/blob/mainnet-staging/node/rest/src/lib.rs#L204.

Reply:
@niklaslong should we consider lowering that limit? 10MB could add up quickly. 

Transactions were the largest thing we accepted and since the size is now capped, does it make sense to allow body sizes greater than 1MB?

Reply:
I don't see why not, I can file another PR for this.

Reply:
Is the 10MB on the request or response size limit?

Remember blocks do not have a maximum size, just like in Ethereum.

Reply:
It _should_ apply only to requests (as it's implemented for the request extractor and not the response).